### DISCUSSION TITLE
Implementing Time Slider for 2D Time Series Data in PyQtGraph

### DISCUSSION THREAD

**Discussion by K-gihu** [pyqtgraph/pyqtgraph#2956] [Q&A]
Hello PyQtGraph community,

I recently came across a feature in the [PyQtGraph documentation](https://pyqtgraph.readthedocs.io/en/latest/getting_started/images.html) which mentions: 

> If the data is 3D (time, x, y), then a time axis will be shown with a slider that can set the currently displayed frame. (if the axes in your data are ordered differently, use numpy.transpose to rearrange them)

However, I have a 2D dataset ordered chronologically (time, x) and would like to similarly display it in PyQtGraph with a time-axis slider that allows users to navigate through the data dynamically. The data is grabbed from a line scan camera, which continuously outputs 1D arrays.

Could someone share an example or recommended approach on how to achieve this with a 1D time series? Any alternative solutions or suggestions to visualize the temporal evolution of my data are greatly appreciated.

Thank you for your insights!

---

I've already implemented a little widget using PlotWidget, QSlider and QSpinBox, by connecting QSlider's valueChanged signal to a function that sets PlotWidget's plot. Nevertheless, I hope to achieve with just pyqtgraph, so suggestions are still welcomed!

![camera](https://github.com/pyqtgraph/pyqtgraph/assets/49723096/b9e882fe-62c2-4883-8fbf-b59da2f61b4d)

**Comment by pijyoi** [comment]
The documentation that you pointed to refers to the `ImageView` widget.
I suppose it could be ab-used to display 1D data.
```python
import numpy as np
import pyqtgraph as pg

data = np.sin(2*np.pi*np.linspace(0, 1, 100))
data = np.expand_dims(data, [1, 2])

pg.mkQApp()
imv = pg.ImageView(discreteTimeLine=True)
imv.show()
imv.setImage(data, axes=dict(t=0, x=1, y=2))
imv.setColorMap(pg.colormap.get("CET-D4"))
imv.ui.histogram.gradient.showTicks(False)
pg.exec()
```

**Reply by K-gihu** [reply]
Sorry for the confusion I made, but I actually want to show a 1D data varying with time, which is a 2D data. I've already edited my description.

**Reply by pijyoi** [reply]
It seems to me that you could visualize your data as an 2D image, with the vertical axis being time.
I do have an application that deals with data in that form.
From memory, the application has 2 `PlotItem`s stacked vertically. An `ImageItem` of the 2D image is added to the lower `PlotItem` and a movable horizontal `InfiniteLine` is placed over the `ImageItem`. A `PlotDataItem` is added to the upper `PlotItem`.
As the user drags the `InfiniteLine`, the corresponding data row that it is over is plotted by the `PlotDataItem`.

**Reply by K-gihu** [reply]
Thank you for mentioning InfiniteLine, I'll try it out.
